usr/bin/cylc

```diff
@@ -147,9 +147,16 @@ fi
 
 if [[ -z "${CYLC_HOME}" ]]; then
     # Look for specified or default version under the root locations
+    # Build CYLC_NAME from CYLC_VERSION if you can:
     if [[ -n "${CYLC_VERSION}" ]]; then
         CYLC_NAME="cylc-$CYLC_VERSION"
-    else
+        echo "cylc name is $CYLC_NAME"
+    fi
+    # Build CYLC_NAME from CYLC_ENV_NAME if you can:
+    if [[ -n "${CYLC_ENV_NAME}" ]]; then
+        CYLC_NAME="${CYLC_ENV_NAME}"
+    fi
+    if [[ -z "${CYLC_ENV_NAME}" && -z "${CYLC_VERSION}" ]]; then
         # Default version symlink
         CYLC_NAME="cylc"
     fi
```

Comment:
We would expect *both* the env name and the version to be specified much of the time so these two need to be exclusive.

The env name should be used in preference to the version.

Reply:
> Note we need to export the env name in this script.

Why?

Reply:
> so these two need to be exclusive.

I thought that I had made them exclusive:
```
if CYLC_VERSION
  use it
if CYLC_ENV_NAME
  use it (regardless of whether you just used it for CYLC_VERSION)
else:
  whateva
```
But I think I've made the logic clearer.